feat: adding GitRepository resource support in flux (#18782)
renovatebot/renovate: lib/modules/manager/flux/readme.md

This manager parses [Flux](https://fluxcd.io/) YAML manifests and supports:

1. [`HelmRelease`](https://fluxcd.io/docs/components/helm/helmreleases/) resources
1. ['GitRepository'](https://fluxcd.io/flux/components/source/gitrepositories/) resources
1. Flux [system](https://fluxcd.io/docs/installation) manifests

### HelmRelease support

Extracts `helm` dependencies from `HelmRelease` resources.

> The `flux` manager will only extract `helm` dependencies for `HelmRelease` resources linked to `HelmRepository` or `GitRepository` sources.

Reviewer: ```suggestion
The `flux` manager only extracts `helm` dependencies for `HelmRelease` resources linked to `HelmRepository` or `GitRepository` sources.
```

> The following configurations are currently unsupported:
> 
> - OCI `HelmRepository` sources (those with `type: oci`)

Reviewer: ```suggestion
Renovate does not support OCI `HelmRepository` sources, those with `type: oci`.
```

A list with one item is a bit silly. Let's rewrite into a single sentence. 😉

Namespaces will not be inferred from the context (e.g. from the parent `Kustomization`).

> As for `HelmRelease` resources that sourced from `GitRepository`, those will be updated by updaing the `GitRepository` resource.

Reviewer: ```suggestion
Renovate updates `HelmRelease` resources coming from `GitRepository` by updating the `GitRepository` resource.
```

We're using `resources` and `sourced` too much. I've tried to rewrite this sentence. I'm still not happy with it, but it's better than before.

Author: Thank you @HonkingGoose , Your sentences are much clearer I like them and happy with them.

### GitRepository support

> Updates `git` references from `GitRepository` resources.

Reviewer: ```suggestion
Renovate can update `git` references from `GitRepository` resources.
```

> The `flux` manager will only update `GitRepository` that have either `tag` or `commit` key.

Reviewer: ```suggestion
The `flux` manager only updates `GitRepository` fields that have a `tag` or `commit` key.
```
